src/ipc/ipc-common.c

```diff
@@ -305,6 +310,21 @@ int ipc_init(struct sof *sof)
 #endif
 
 #ifdef __ZEPHYR__
+	struct k_thread *thread = &ipc_send_wq.thread;
+
+	k_work_queue_start(&ipc_send_wq,
+		       ipc_send_wq_stack,
+		       K_THREAD_STACK_SIZEOF(ipc_send_wq_stack),
+		       1, NULL);
+
+	k_thread_suspend(thread);
+
+	k_thread_cpu_mask_clear(thread);
+	k_thread_cpu_mask_enable(thread, PLATFORM_PRIMARY_CORE_ID);
+	k_thread_name_set(thread, "ipc_send_wq");
+
+	k_thread_resume(thread);
```

Comment:
@lyakh probably worth a comment describing the aim here since its more than a couple of lines of code.
@peter-mitsis @andyross is there a convenience API for the above "pin work to a single core" ?

Reply:
a slight improvement would be to use `k_thread_cpu_pin()`